aggsender/types/interfaces.go

```diff
@@ -177,13 +177,44 @@ type VerifyIncomingRequest struct {
 	PreviousCertificate *agglayertypes.CertificateHeader
 }
 
+// HealthCheckStatus defines the status of a health check
+type HealthCheckStatus = string
+
+const (
+	HealthCheckStatusOK HealthCheckStatus = "OK"
+)
+
+// HealthCheckResponse response for health check
+type HealthCheckResponse struct {
+	Status       HealthCheckStatus
+	StatusReason string
+	Version      string
+}
+
+// IsHealthy checks if the health check response is healthy
+func (h *HealthCheckResponse) IsHealthy() bool {
+	return h != nil && h.Status == HealthCheckStatusOK
+}
+
+// String returns a string representation of the HealthCheckResponse
+func (h *HealthCheckResponse) String() string {
+	if h == nil {
+		return "HealthCheckResponse is nil"
+	}
+	return "HealthCheckResponse{Status: " + h.Status +
+		", StatusReason: " + h.StatusReason +
+		", Version: " + h.Version + "}"
+}
+
 type CertificateValidator interface {
 	ValidateCertificate(ctx context.Context, params VerifyIncomingRequest) error
 }
 
 // CertificateValidateAndSigner is an interface to attach a certificate validator and signer
 // to aggsender regular flow
 type CertificateValidateAndSigner interface {
```

Comment:
WDYT, more concise IMO?

```suggestion
type CertificateValidator interface {
```

Reply:
`CertificateValidator` already exists, it just validate but don't sign the certificate